app/src/behaviors/behavior_sensor_rotate_var.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2022 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_behavior_sensor_rotate_var
+
+#include <device.h>
+#include <drivers/behavior.h>
+#include <logging/log.h>
+
+#include <drivers/sensor.h>
+#include <zmk/event_manager.h>
+#include <zmk/events/keycode_state_changed.h>
+#include <zmk/behavior_queue.h>
+#include <zmk/keymap.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+struct behavior_sensor_rotate_var_config {
+    char *cw_behavior_dev;
+    char *ccw_behavior_dev;
+    int default_wait_ms;
+};
+
+static int on_sensor_binding_triggered(struct zmk_behavior_binding *binding,
+                                       const struct device *sensor, int64_t timestamp) {
+    const struct device *dev = device_get_binding(binding->behavior_dev);
+    const struct behavior_sensor_rotate_var_config *cfg = dev->config;
+
+    struct sensor_value value;
+    int err;
+
+    err = sensor_channel_get(sensor, SENSOR_CHAN_ROTATION, &value);
+
+    if (err) {
+        LOG_WRN("Failed to get sensor rotation value: %d", err);
+        return err;
+    }
+
+    struct zmk_behavior_binding triggered_binding;
+    switch (value.val1) {
+    case 1:
+        triggered_binding.behavior_dev = cfg->cw_behavior_dev;
+        triggered_binding.param1 = binding->param1;
+        break;
+    case -1:
+        triggered_binding.behavior_dev = cfg->ccw_behavior_dev;
+        triggered_binding.param1 = binding->param2;
+        break;
+    default:
+        return -ENOTSUP;
+    }
+
+    LOG_DBG("Sensor binding: %s", log_strdup(binding->behavior_dev));
+
+    zmk_behavior_queue_add(0, triggered_binding, true, 0);
+    zmk_behavior_queue_add(0, triggered_binding, false, cfg->default_wait_ms);
```

Comment:
I'm not convinced this is actually doing what you want. It seems like this really should be:

```suggestion
    zmk_behavior_queue_add(0, triggered_binding, true, cfg->tap_ms);
    zmk_behavior_queue_add(0, triggered_binding, false, 0);
```